tests/ui/traits/question-mark-result-err-mismatch.stderr

```diff
@@ -0,0 +1,83 @@
+error[E0277]: `?` couldn't convert the error to `String`
+  --> $DIR/question-mark-result-err-mismatch.rs:12:22
+   |
+LL |   fn foo() -> Result<String, String> {
+   |               ---------------------- expected `String` because of this
+...
+LL |           .ok_or_else(|| {
+   |  __________-
+LL | |             "Couldn't split the test string"
+LL | |         });
+   | |__________- this can be annotated with `?` because it has type `Result<&str, &str>`
+LL |       let one = x
+LL |           .map(|s| ())
+   |            ----------- this can be annotated with `?` because it has type `Result<(), &str>`
+LL |           .map_err(|_| ())
+   |            --------------- this can't be annotated with `?` because it has type `Result<(), ()>`
+LL |           .map(|()| "")?;
+   |            ------------^ the trait `From<()>` is not implemented for `String`
+   |            |
+   |            this can't be annotated with `?` because it has type `Result<&str, ()>`
```

Comment:
this seems really heavy for long chains -- do i really care about `?`-ability 4 `.map`s down?

Reply:
Probably not. I would like to add an extra check to only point at places where the `E` *changed* (like we do for assoc types in method call chains), but wanted to keep the code down to a reviewable size.

Reply:
I'd rather just put a hard limit on the number of method chain to walk backwards. Maybe 2 is sensible.

Reply:
Let me know what you think of the new output.